lib/winapi/filemanip.c

```diff
@@ -235,3 +239,80 @@ BOOL MoveFileA (LPCTSTR lpExistingFileName, LPCTSTR lpNewFileName)
         return TRUE;
     }
 }
+
+DWORD GetLogicalDrives (VOID)
+{
+    DWORD result = 0;
+    NTSTATUS status;
+    ANSI_STRING path;
+    HANDLE handle;
+    OBJECT_ATTRIBUTES attributes;
+    struct {
+        OBJECT_DIRECTORY_INFORMATION objDirInfo;
+        CHAR filenameBuf[100];
+    } objDirInfoBuf;
+
+    InitializeObjectAttributes(&attributes, NULL, OBJ_CASE_INSENSITIVE, ObDosDevicesDirectory(), NULL);
+    status = NtOpenDirectoryObject(&handle, &attributes);
+
+    if (!NT_SUCCESS(status)) {
+        SetLastError(RtlNtStatusToDosError(status));
+        return 0;
+    }
+
+    for (ULONG index = 0;;) {
```

Comment:
nit: This feels like a `while` loop to me.

Reply:
I tried to keep the scope of `index` small, but it's changed now.